daemon/exec.go

```diff
@@ -248,3 +248,34 @@ func (d *Daemon) Exec(c *Container, execConfig *execConfig, pipes *execdriver.Pi
 
 	return exitStatus, err
 }
+
+// execCommandGC starts a go routine on a ticker to clean up the daemon references
+// of exec configs that are no longer part of the container.
+func (d *Daemon) execCommandGC() {
+	for range time.Tick(5 * time.Minute) {
+		var (
+			cleaned          int
+			liveExecCommands = d.containerExecIds()
+			ids              = d.execCommands.List()
+		)
+		for _, id := range ids {
+			if _, exists := liveExecCommands[id]; !exists {
```

Comment:
Because I get the container ids first before I itterate over the daemon objects this should not make the race happen because both the List calls are behind a mutex.  If the command starts and exits the lock on modifying the list will keep this from happening and it will still be viewed as alive for that tick.